elastic/kibana
# [RAC] Fix rule registry write flag and turn it off by default

## :memo: Summary

This changes the default of `xpack.ruleRegistry.write.enabled` to `false`. It also fixes the setting to actually prevent writing of assets and data.

## :detective: Review notes

- On a shared cluster the assets must be namespaced using `xpack.ruleRegistry.index`.
- Without `xpack.ruleRegistry.write.enabled` no component and index templates are allowed to be created at startup.
- Without `xpack.ruleRegistry.write.enabled` no indices and documents are allowed to be created during alert execution.
- Currently the APM alerts and the experimental Security solution alerts (enabled via `xpack.securitySolution.enableExperimental: ['ruleRegistryEnabled']` should be affected.

## :test_tube: Testing

- with `write: false`:
  - check that no rule data assets (`.alerts-*`) are created at startup
  - check that no rule data indices (`.alerts-*`) are created at rule check execution
  - check that relevant alerts still function
- with `write: true`:
  - check that rule data assets (`.alerts-*`) are created at startup
  - check that rule data indices (`.alerts-*`) are created at rule check execution
  - check that relevant alerts still function
